backend/core/src/main/java/org/sonarsource/sonarlint/core/sca/ScaService.java

```diff
@@ -90,6 +103,38 @@ private static ServerScaIssue.Transition adaptTransition(DependencyRiskTransitio
     };
   }
 
+  public void openDependencyRiskInBrowser(String configurationScopeId, UUID dependencyKey) {
+    var effectiveBinding = configurationRepository.getEffectiveBinding(configurationScopeId);
+    var endpointParams = effectiveBinding.flatMap(binding -> connectionRepository.getEndpointParams(binding.connectionId()));
+    if (effectiveBinding.isEmpty() || endpointParams.isEmpty()) {
+      LOG.warn("Configuration scope {} is not bound properly, unable to open dependency risk", configurationScopeId);
+      return;
+    }
+    var branchName = branchTrackingService.awaitEffectiveSonarProjectBranch(configurationScopeId);
+    if (branchName.isEmpty()) {
+      LOG.warn("Configuration scope {} has no matching branch, unable to open dependency risk", configurationScopeId);
```

Comment:
I know it's done the same way for hotspots, but I think having a RPC notification is maybe not so great for this use case. If a precondition is not met, the only thing we can do is log. We could use `showMessage`, but it kind of constraints the UI. If it was a request, we could fail it, and let clients handle the error as they see fit. This can also be addressed via the errors UX effort that should happen soon